src/models/User.ts

```diff
@@ -283,6 +285,8 @@ userSchema.plugin(mongoosePaginate);
 const userModel = (): PaginateModel<InterfaceUser> =>
   model<InterfaceUser, PaginateModel<InterfaceUser>>("User", userSchema);
 
+createLoggingMiddleware(userSchema, "User");
```

Comment:
This is so elegant! Thanks for putting in the work to research the same!

Reply:
Thank you @EshaanAgg

Reply:
Should I add this middleware in all model files?

Reply:
No. Let's merge this and wait a few days so all the developers can try it. If we get a positive feedback, and there were no unexpected errors encountered, then we can work on adding the same for all models.